lib/core/src/system/web/text_input.rs

```diff
@@ -0,0 +1,130 @@
+
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        pub fn stop_handling(this:&TextInputHandlers);
+    }
+}
+
+
+
+// =======================
+// === KeyboardBinding ===
+// =======================
+
+/// Copy handler takes bool which is true on cut operations, and returns the string should be
+/// actually copied to clipboard.
+pub trait CopyHandler          = FnMut(bool) -> String + 'static;
+
+/// The paste handler takes in its argument the text from clipboard to paste.
+pub trait PasteHandler         = FnMut(String) + 'static;
+
+/// Keyboard event handler takes event as an argument.
+pub trait KeyboardEventHandler = FnMut(KeyboardEvent) + 'static;
+
+/// The keyboard event bindings.
+///
+/// This structure wraps the javascript content handling events in a way describing in this module
+/// docs.
+pub struct KeyboardBinding {
+    js_handlers      : js::TextInputHandlers,
+    copy_handler     : Option<Closure<dyn CopyHandler>>,
+    paste_handler    : Option<Closure<dyn PasteHandler>>,
+    key_down_handler : Option<Closure<dyn KeyboardEventHandler>>,
+    key_up_handler   : Option<Closure<dyn KeyboardEventHandler>>,
+}
+
+impl KeyboardBinding {
+    /// Add the textarea element to document body and returns KeyboardBinding structure which uses
+    /// it, without any handlers.
+    pub fn create() -> Self {
+        KeyboardBinding {
+            js_handlers      : js::TextInputHandlers::new(),
+            copy_handler     : None,
+            paste_handler    : None,
+            key_down_handler : None,
+            key_up_handler   : None
+        }
+    }
+
+    /// Set copy handler.
+    pub fn set_copy_handler<Handler:CopyHandler>(&mut self, handler:Handler) {
+        let handler_js : Closure<dyn CopyHandler> = Closure::wrap(Box::new(handler));
+        self.js_handlers.set_copy_handler(&handler_js);
+        self.copy_handler = Some(handler_js);
+    }
+
+    /// Set paste handler.
+    pub fn set_paste_handler<Handler:PasteHandler>(&mut self, handler:Handler) {
+        let handler_js : Closure<dyn PasteHandler> = Closure::wrap(Box::new(handler));
+        self.js_handlers.set_paste_handler(&handler_js);
+        self.paste_handler = Some(handler_js);
+    }
+
+    /// Set keydown handler.
+    pub fn set_key_down_handler<Handler:KeyboardEventHandler>(&mut self, handler:Handler) {
+        let handler_js : Closure<dyn KeyboardEventHandler> = Closure::wrap(Box::new(handler));
+        self.js_handlers.set_event_handler("keydown", &handler_js);
+        self.key_down_handler = Some(handler_js);
+    }
+
+    /// Set keyup handler.
+    pub fn set_key_up_handler<Handler:KeyboardEventHandler>(&mut self, handler:Handler) {
+        let handler_js : Closure<dyn KeyboardEventHandler> = Closure::wrap(Box::new(handler));
+        self.js_handlers.set_event_handler("keyup", &handler_js);
+        self.key_up_handler = Some(handler_js);
+    }
```

Comment:
Make a simple local macro for this repeating code. The macro would be as readible as this code but the code would be 4 times shorter.

Reply:
Not quiet. There is a little difference between `set_paste_handler` and `set_keyfoo_handler` in a js_handlers method call. And that little differences makes macro less readable IMO. 4 times shorter not always means 4 times faster to read.